app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ReservationPresenter.java

```diff
@@ -1,16 +1,56 @@
 package com.example.parkingsystemjava.mvp.presenter;
 
+import com.example.parkingsystemjava.ReservationObject.Reservation;
 import com.example.parkingsystemjava.mvp.contract.ReservationActivityContract;
+import java.util.Calendar;
 
 public class ReservationPresenter implements ReservationActivityContract.PresenterContract {
     private ReservationActivityContract.ViewContract view;
+    private ReservationActivityContract.ModelContract model;
 
-    public ReservationPresenter(ReservationActivityContract.ViewContract view) {
+    public ReservationPresenter(ReservationActivityContract.ViewContract view, ReservationActivityContract.ModelContract model) {
         this.view = view;
+        this.model = model;
     }
 
     @Override
-    public void dateTimePickerButtonPressed() {
-        view.showDateTimeInfo();
+    public void dateTimePickerButtonStartPressed() {
+        view.showDateTimeInfo(true);
+    }
+
+    @Override
+    public void dateTimePickerButtonEndPressed() {
+        view.showDateTimeInfo(false);
+    }
+
+    @Override
+    public void setReservationCalendar(Calendar date, boolean isStartDate) {
+        if (isStartDate) {
+            model.setStartDate(date);
+        } else {
+            model.setEndDate(date);
+        }
+    }
+
+    @Override
+    public void buttonSaveReservation() {
+        model.saveReservation(view.getParkingLots(), view.getPass());
+        Reservation reservation = model.getReservation(view.getParkingLots(), view.getPass());
+        if (check(reservation)) {
+            view.showMessageOfSavedStatement(reservation.getStartDate(), reservation.getEndDate(), reservation.getParkingLots(), reservation.getPass());
```

Comment:
why you don't send the complete object?

Reply:
Done, changed. I send the complete object now